### PR TITLE
DBZ-4900 Use exponential back-off for log availability check

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-4900

### CODE REVIEW

**Review by gunnarmorling — COMMENTED** [review]
@Naros, LGTM, the only thing that should be changed consistently is to use `Duration` rather than `long` ms. As discussed before, the semantic Java 8 time/date types should be used internally as much as possible, only at the system boundary (i.e. when consuming config options or when calling external APIs not aware of `Duration`) the value should be converted.

**Comment by jpechane** [comment]
@gunnarmorling I've added a commit using Duration so please take a look

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 1]
```diff
@@ -49,34 +48,37 @@
...
+                                                        boolean archiveLogOnlyMode, String archiveDestinationName, int maxRetries,
+                                                        Duration initialDelay, Duration maxDelay)
+            throws SQLException {
         List<BigInteger> ret = new LinkedList<>();
         removeLogFilesFromMining(connection);
 
         // Restrict max attempts to 0 or greater values (sanity-check)
         // the code will do at least 1 attempt and up to maxAttempts extra polls based on configuration
         final int maxAttempts = Math.max(maxRetries, 0);
+        final DelayStrategy retryStrategy = DelayStrategy.exponential(initialDelay.toMillis(), maxDelay.toMillis());
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:67]: We should add an overload for `exponential()` to accept `Duration` too (or better yet, just change it). Can happen separately though.

**Comment by gunnarmorling** [comment]
LGTM now, merging. Thx for the quick update, @jpechane, and the original work, @Naros!

**Comment by gunnarmorling** [comment]
Logged https://issues.redhat.com/browse/DBZ-4902 as a follow-up. Probably a matter of minutes to do this change.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:67]: Well, it is a separate task. *All* factory methods need to be updated.

Comment (gunnarmorling) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:67]: Yep, it's tracked via a follow-up issue now.

**Pull request merged**